src/domain/accounts/counterfactual-safes/entities/create-counterfactual-safe.dto.entity.ts

```diff
@@ -0,0 +1,34 @@
+import { AddressSchema } from '@/validation/entities/schemas/address.schema';
+import { z } from 'zod';
+
+export class CreateCounterfactualSafeDto
+  implements z.infer<typeof CreateCounterfactualSafeDtoSchema>
+{
+  chain_id: string;
+  fallback_handler: `0x${string}`;
+  owners: `0x${string}`[];
+  predicted_address: `0x${string}`;
+  salt_nonce: string;
+  singleton_address: `0x${string}`;
+  threshold: number;
```

Comment:
Nit: I think we should remain consistent and expect camelcase.

Reply:
I agree, fixed in [485ece5](https://github.com/safe-global/safe-client-gateway/pull/1773/commits/485ece55c7f0ab4f2c494a1ea66156f46db630e3)

My initial thought was to use snake_case in the domain DTO and camelCase in the route DTO. This would align better with the fact we are returning snake_case entities in the domain (so the DTO would be snake_case also), but it makes no sense with the current naming standard, which seems to be:

**Input (DTO)**
Route: camelCase
Domain: camelCase
Datasource: here the mapping camelCase-> snake_case is done

**Output (result entity)**
Datasource: snake_case
Domain: snake_case
Route: here the mapping snake_case -> camelCase is done

What do you think about mapping both DTO and entities in the same layer? I'd say the datasource seems the proper place. That would restrict the usage of snake_case to the datasource only, and the domain would only see camelCase for both DTOs and entities.

So the proposal would be:

Input (DTO)
Route: camelCase
Domain: camelCase
Datasource: here the mapping camelCase-> snake_case is done

Output (result entity)
Datasource: here the mapping snake_case -> camelCase is done
Domain: camelCase
Route: camelCase

If you are happy with the proposal, I'll take care of it in a separate PR.

Reply:
I agree with the proposal!